sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuReadParquetFileFormat.scala

```diff
@@ -57,7 +57,20 @@ class GpuReadParquetFileFormat extends ParquetFileFormat with GpuReadFileFormatW
 }
 
 object GpuReadParquetFileFormat {
+
+  private val hiddenMetadataFields = Set(
```

Comment:
Rather than manually maintain a list of metadata columns and hope we're keeping it in sync with Spark, I think it would be a lot easier to just check the `metadata` field of the original `FileSourceScan` (in Spark 3.3.x+) and make sure we do not convert it to the GPU if it is non-empty.  Spark is already determining whether there are metadata fields, and we can reuse that logic to stay in sync.